lib/msf/ui/console/command_dispatcher/core.rb

```diff
@@ -1650,9 +1666,25 @@ def cmd_sessions(*args)
         end
       end
     when 'killall'
-      print_status("Killing all sessions...")
-      framework.sessions.each_sorted.reverse_each do |s|
-        session = framework.sessions.get(s)
+      if search_term
+        matching_sessions = get_matching_sessions(search_term)
+        if matching_sessions.empty?
+          print_status("No matching sessions.") 
+          return
+        end
+        print_status("Killing matching sessions...")
+        print_line
+        print(Serializer::ReadableText.dump_sessions(framework, show_active: show_active, show_inactive: show_inactive, show_extended: show_extended, verbose: verbose, session_ids: matching_sessions.keys()))
+        print_line
+      else
+        matching_sessions = framework.sessions
+        print_status("Killing all sessions...")
```

Comment:
```suggestion
        matching_sessions = framework.sessions
        print_status("Killing all sessions...")
```
In conjunction with previous comment this can be moved further up

Reply:
Leaving this here since knowing if this is the result of a search is specifically useful for killall output, but let me know if I'm misinterpreting